### PR TITLE
Add config option for iptables backend

### PR DESCRIPTION
**Pull request by tmjd**
## Release Note

```release-note
None required
```

### CODE REVIEW

File: lib/validator/v3/validator.go [hunk 1]
```diff
@@ -278,6 +279,12 @@ func validateMAC(fl validator.FieldLevel) bool {
 	return true
 }
 
+func validateIptablesBackend(fl validator.FieldLevel) bool {
+	s := fl.Field().String()
+	log.Debugf("Validate Iptables Backend: %s", s)
+	return s == "" || s == "nft" || s == "legacy"
```
Comment (caseydavenport) [inline lib/validator/v3/validator.go]: For k8s resources the convention is that named values start with an uppercase.

e.g.,

`iptablesBackend: Legacy` or `iptablesBackend: NFTables` 

We probably want to use a type with some constants for this as well.

```
type IPTablesBackend string
```

File: lib/apis/v3/felixconfig.go [hunk 2]
```diff
@@ -246,6 +246,9 @@ type FelixConfigurationSpec struct {
 	// modes can use XDP. This is not recommended since it doesn't provide better performance than
 	// iptables. [Default: false]
 	GenericXDPEnabled *bool `json:"genericXDPEnabled,omitempty" confignamev1:"GenericXDPEnabled"`
+
+	// IPtablesBackend
+	IptablesBackend string `json:"iptablesBackend,omitempty" validate:"omitempty,iptablesBackend"`
```
Comment (caseydavenport) [inline lib/apis/v3/felixconfig.go]: I believe it should be `IPTablesBackend`

Comment (tmjd) [reply lib/apis/v3/felixconfig.go]: I'm fine with that but I feel like the command is always `iptables`, no capitalization, so I thought just the 1st I should be capitalized (even though I was inconsistent with that).

Comment (caseydavenport) [reply lib/apis/v3/felixconfig.go]: Yeah, I can see that. I guess it doesn't matter much.

File: lib/apis/v3/felixconfig.go [hunk 3]
```diff
@@ -246,6 +250,9 @@ type FelixConfigurationSpec struct {
 	// modes can use XDP. This is not recommended since it doesn't provide better performance than
 	// iptables. [Default: false]
 	GenericXDPEnabled *bool `json:"genericXDPEnabled,omitempty" confignamev1:"GenericXDPEnabled"`
+
+	// IPTablesBackend specifies which backend of iptables will be used. The default is legacy.
+	IPTablesBackend *IPTablesBackend `json:"iptablesBackend,omitempty" validate:"omitempty,iptablesBackend"`
```
Comment (fasaxc) [inline lib/apis/v3/felixconfig.go]: There are other iptables config params further up, I think this should be in the same "block" as those and it should follow the same casing.  For better or worse, the existing ones are "IptablesXXX"  (Think I followed the casing used by the iptables team, but I can't say it looks good!)

**Pull request merged**